Not compatible with Keycloak 23

Keycloak 23 was released on 23.11.2023 and [removes the `RegistrationProfile` form action](https://www.keycloak.org/docs/latest/upgrading/index.html#removed-registrationprofile-form-action) in favor of `RegistrationUserCreation`. Because `RegistrationProfile` is used by this plugin, running `./kc.sh build` with this plugin fails on Keycloak 23:
```
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]: ERROR: Build failure: Build failed due to errors
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         ... 30 more
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:516)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:518)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]: Caused by: java.lang.ClassNotFoundException: org.keycloak.authentication.forms.RegistrationProfile
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.Thread.run(Thread.java:840)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:864)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.keycloak.quarkus.deployment.KeycloakProcessor.configureKeycloakSessionFactory(KeycloakProcessor.java:370)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.keycloak.quarkus.deployment.KeycloakProcessor.loadFactories(KeycloakProcessor.java:673)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.keycloak.provider.ProviderManager.load(ProviderManager.java:93)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at org.keycloak.provider.DefaultProviderLoader.load(DefaultProviderLoader.java:60)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1228)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1217)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.Class.forName(Class.java:467)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.Class.forName0(Native Method)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:506)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.ClassLoader.defineClass1(Native Method)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:506)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         at java.base/java.lang.ClassLoader.defineClass1(Native Method)
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]:         [error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureKeycloakSessionFactory threw an exception: java.lang.NoClassDefFoundError: org/keycloak/authentication/forms/RegistrationProfile
Dec 04 21:47:43 ubuntu-jammy kc.sh[2164]: ERROR: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
```

Will this plugin be updated to be compatible to Keycloak 23?
I'm looking into migrating to `RegistrationUserCreation` for another plugin, maybe I can give some hints after I figured it out.

--- a/pom.xml
+++ b/pom.xml
@@ -32,7 +32,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>17</maven.compiler.source>
     <maven.compiler.target>17</maven.compiler.target>
-    <keycloak.version>22.0.0</keycloak.version>
+    <keycloak.version>23.0.3</keycloak.version>
   </properties>
 
   <dependencies>

--- a/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationDomainValidation.java
+++ b/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationDomainValidation.java
@@ -2,9 +2,10 @@ package net.micedre.keycloak.registration;
 
 import org.jboss.logging.Logger;
 import org.keycloak.authentication.FormAction;
+import org.keycloak.authentication.FormContext;
 import org.keycloak.authentication.ValidationContext;
 import org.keycloak.authentication.forms.RegistrationPage;
-import org.keycloak.authentication.forms.RegistrationProfile;
+import org.keycloak.authentication.forms.RegistrationUserCreation;
 import org.keycloak.events.Details;
 import org.keycloak.events.Errors;
 import org.keycloak.models.AuthenticatorConfigModel;
@@ -16,8 +17,8 @@ import jakarta.ws.rs.core.MultivaluedMap;
 import java.util.ArrayList;
 import java.util.List;
 
-public abstract class RegistrationProfileDomainValidation extends RegistrationProfile implements FormAction {
-   protected static final Logger logger = Logger.getLogger(RegistrationProfileDomainValidation.class);
+public abstract class RegistrationUserCreationDomainValidation extends RegistrationUserCreation implements FormAction {
+   protected static final Logger logger = Logger.getLogger(RegistrationUserCreationDomainValidation.class);
 
    protected static final String DEFAULT_DOMAIN_LIST = "example.org";
    protected static final String DOMAIN_LIST_SEPARATOR = "##";
@@ -25,6 +26,10 @@ public abstract class RegistrationProfileDomainValidation extends RegistrationPr
    @Override
    public boolean isConfigurable() {
         return true;
+   }
+
+   @Override
+   public void success(FormContext context) {
    }
 
    protected static final boolean globmatches(String text, String glob) {

--- a/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationWithDomainBlock.java
+++ b/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationWithDomainBlock.java
@@ -9,7 +9,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
-public class RegistrationProfileWithDomainBlock extends RegistrationProfileDomainValidation {
+public class RegistrationUserCreationWithDomainBlock extends RegistrationUserCreationDomainValidation {
 
    public static final String PROVIDER_ID = "registration-domain-block-action";
 

--- a/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationWithMailDomainCheck.java
+++ b/src/main/java/net/micedre/keycloak/registration/RegistrationUserCreationWithMailDomainCheck.java
@@ -9,7 +9,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
-public class RegistrationProfileWithMailDomainCheck extends RegistrationProfileDomainValidation {
+public class RegistrationUserCreationWithMailDomainCheck extends RegistrationUserCreationDomainValidation {
 
    public static final String PROVIDER_ID = "registration-mail-check-action";
 

--- a/src/main/resources/META-INF/services/org.keycloak.authentication.FormActionFactory
+++ b/src/main/resources/META-INF/services/org.keycloak.authentication.FormActionFactory
@@ -1,2 +1,2 @@
-net.micedre.keycloak.registration.RegistrationProfileWithMailDomainCheck
-net.micedre.keycloak.registration.RegistrationProfileWithDomainBlock
+net.micedre.keycloak.registration.RegistrationUserCreationWithMailDomainCheck
+net.micedre.keycloak.registration.RegistrationUserCreationWithDomainBlock

